packages/firestore/test/integration/api/database.test.ts

```diff
@@ -1229,4 +1230,111 @@ apiDescribe('Database', (persistence: boolean) => {
+      }
+    };
+
+    it('for DocumentReference.withConverter()', () => {
+      return withTestDb(persistence, async db => {
+        const docRef = db
+          .collection('posts')
+          .doc()
+          .withConverter(postConverter);
+
+        await docRef.set(new Post('post', 'author'));
+        const postData = await docRef.get();
+        const post = postData.data();
+        expect(post).to.not.equal(undefined);
+        expect(post!.byline()).to.equal('post, by author');
+      });
+    });
+
+    it('for CollectionReference.withConverter()', () => {
+      return withTestDb(persistence, async db => {
+        const docRef = db
+          .collection('posts')
+          .withConverter(postConverter)
+          .doc();
+
+        await docRef.set(new Post('post', 'author'));
+        const postData = await docRef.get();
+        const post = postData.data();
+        expect(post).to.not.equal(undefined);
+        expect(post!.byline()).to.equal('post, by author');
+      });
+    });
+
+    it('for Query.withConverter()', () => {
+      return withTestDb(persistence, async db => {
+        await db
+          .doc('postings/post1')
+          .set({ title: 'post1', author: 'author1' });
+        await db
+          .doc('postings/post2')
+          .set({ title: 'post2', author: 'author2' });
+        const posts = await db
+          .collectionGroup('postings')
+          .withConverter(postConverter)
+          .get();
+        expect(posts.size).to.equal(2);
+        expect(posts.docs[0].data()!.byline()).to.equal('post1, by author1');
+      });
+    });
+
+    it('calls DocumentSnapshot.data() with specified SnapshotOptions', () => {
+      return withTestDb(persistence, async db => {
+        const docRef = db.doc('some/doc').withConverter({
+          toFirestore(post: Post): firestore.DocumentData {
+            return { title: post.title, author: post.author };
+          },
+          fromFirestore(
+            snapshot: firestore.QueryDocumentSnapshot,
+            options: firestore.SnapshotOptions
+          ): Post {
+            // Check that options were passed in properly.
+            expect(options).to.deep.equal({ serverTimestamps: 'estimate' });
+
+            const data = snapshot.data(options);
+            return new Post(data.title, data.author);
+          }
+        });
+
+        await docRef.set(new Post('post', 'author'));
+        const postData = await docRef.get();
+        postData.data({ serverTimestamps: 'estimate' });
+      });
+    });
+
+    it('drops the converter when calling CollectionReference<T>.parent()', () => {
+      return withTestDb(persistence, async db => {
+        const postsCollection = db
+          .collection('users/user1/posts')
+          .withConverter(postConverter);
+
+        const usersCollection = postsCollection.parent;
+        expect((usersCollection as DocumentReference)._converter).to.be
+          .undefined;
```

Comment:
Note that accessing `_` properties from tests is kinda' bad practice and it may break if @schmidt-sebastian gets his property minification stuff to work. 😛 

So you might consider finding another way to write this...  Perhaps use `isEqual()`?